pkg/cloud/cloud.go

```diff
@@ -269,13 +271,26 @@ func (c *cloud) CreateDisk(ctx context.Context, volumeName string, diskOptions *
 			return nil, fmt.Errorf("failed to get availability zone %s", err)
 		}
 	}
-
-	request := &ec2.CreateVolumeInput{
-		AvailabilityZone:  aws.String(zone),
-		Size:              aws.Int64(capacityGiB),
-		VolumeType:        aws.String(createType),
-		TagSpecifications: []*ec2.TagSpecification{&tagSpec},
-		Encrypted:         aws.Bool(diskOptions.Encrypted),
+	var request *ec2.CreateVolumeInput
+
+	// EBS doesn't handle empty outpost arn, so we have to include it only when it's non-empty
+	if len(diskOptions.OutpostArn) > 0 {
```

Comment:
could declare request outside this if and then set request.OutpostArn in the if to reduce some of the duplication.